pkg/antctl/raw/multicluster/join.go

```diff
@@ -0,0 +1,288 @@
+				   --leader-clusterid=cluster-north \
+				   --leader-apiserver=https://172.18.0.3:6443 \
+				   --clusterset=clusterset \
+				   --namespace=kube-system \
+				   --clusterid=cluster-east \
+				   --token-secret-file=member-access-token.yaml
+
+# Join the ClusterSet from a member cluster with a config file. 
+  $ antctl mc join --config-file ./config.yaml
+
+# Config file example:
+apiVersion: multicluster.antrea.io/v1alpha1
+kind: ClusterSetJoinConfig
+namespace: kube-system
+clusterID: cluster-east
+leaderNamespace: antrea-multicluster
+tokenSecretName: member-access-token
+leaderAPIServer: https://172.18.0.3:6443
+leaderClusterID: cluster-north
+clusterSetID: clusterset
+`, "\n")
+
+func NewJoinCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "join",
+		Short:   "Join the ClusterSet from a member cluster",
+		Args:    cobra.MaximumNArgs(0),
+		Example: joinExamples,
+		RunE:    joinRunE,
+	}
+
+	o := JoinOptions{}
+	joinOpts = &o
+	command.Flags().StringVarP(&joinOpts.LeaderNamespace, "leader-namespace", "", "", "Namespace of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterID, "leader-clusterid", "", "", "Cluster ID of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.TokenSecretName, "token-secret-name", "", "", "Name of the Secret resource that contains the leader cluster token")
+	command.Flags().StringVarP(&joinOpts.LeaderAPIServer, "leader-apiserver", "", "", "API Server endpoint of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.Namespace, "namespace", "n", "", "Antrea Multi-cluster Namespace")
+	command.Flags().StringVarP(&joinOpts.ClusterID, "clusterid", "", "", "Cluster ID of the member cluster")
+	command.Flags().StringVarP(&joinOpts.ClusterSetID, "clusterset", "", "", "ClusterSet ID")
+	command.Flags().StringVarP(&joinOpts.TokenSecretFile, "token-secret-file", "", "", "Secret file to the leader cluster")
+	command.Flags().StringVarP(&joinOpts.ConfigFile, "config-file", "f", "", "Config file that defines all config options. If both command line options and config file are specified, "+
+		"the arguments in config file will be used.")
+
+	return command
+}
+
+func joinRunE(cmd *cobra.Command, args []string) error {
+	var err error
+	if err = joinOpts.validateAndComplete(); err != nil {
+		return err
+	}
+	k8sClient, err := common.NewClient(cmd)
+
+	memberClusterNamespace := joinOpts.Namespace
+	memberClusterID := joinOpts.ClusterID
+	memberClusterSet := joinOpts.ClusterSetID
+	createdRes := []map[string]interface{}{}
+	defer func() {
+		if err != nil {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to join the ClusterSet. Deleting the created resources\n")
+			if err := common.Rollback(cmd, k8sClient, createdRes); err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "Failed to rollback: %v\n", err)
+			}
+		}
+	}()
+
+	// The command prefers to use the Secret in the config file, but if the Secret is not specified. It will create a Secret from the config file.
```

Comment:
I feel the precedence order should be: TokenSecretName > TokenSecretFile > Secret in the config file. So, users have a chance to override the default token in the config file.

Reply:
Done.